osu.Game/Beatmaps/Drawables/DifficultyIconTooltip.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osuTK;
+
+namespace osu.Game.Beatmaps.Drawables
+{
+    public class DifficultyIconTooltip : VisibilityContainer, ITooltip<DifficultyIconTooltipContent>
```

Comment:
Probably should be `internal` at least rather than `public`.

Reply:
keep in mind this may break dynamic compilation in weird ways.

Reply:
I thought internals are safe for dynamic compilation now? I guess it's not a huge deal but it would be nice to not outwardly expose this.

Reply:
Is it? https://github.com/ppy/osu-framework/issues/3727 is still open and I don't recall anything happening regarding it.

Reply:
https://github.com/ppy/osu-framework/pull/4006 this one happened after.